applications/FemToDemApplication/custom_processes/compute_normalized_free_energy_on_nodes_process.cpp

```diff
@@ -0,0 +1,335 @@
+
+void ComputeNormalizedFreeEnergyOnNodesProcess::Execute()
+{
+    int max_id;
+    this->ObtainMaximumNodeId(max_id);
+    NodeNormalizedFreeEnergy *NodeNormalizedFreeEnergyVector = new NodeNormalizedFreeEnergy[max_id];
+    this->NormalizedFreeEnergyExtrapolation(NodeNormalizedFreeEnergyVector);
+    delete[] NodeNormalizedFreeEnergyVector;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ComputeNormalizedFreeEnergyOnNodesProcess::NormalizedFreeEnergyExtrapolation(
+    NodeNormalizedFreeEnergy *pNodeNormalizedFreeEnergyVector
+    )
+{
+    Vector gauss_point_stress;
+    double gauss_point_damage;
+    // Loop over elements to extrapolate the stress to the nodes
+    for (auto it_elem = mrModelPart.Elements().ptr_begin(); it_elem != mrModelPart.Elements().ptr_end(); ++it_elem) {
+        auto& r_geometry = (*it_elem)->GetGeometry();
+        auto& r_mat_properties = (*it_elem)->GetProperties();
+
+        const Vector& r_strain_vector = (*it_elem)->GetValue(STRAIN_VECTOR);
+        const Vector& r_stress_vector = (*it_elem)->GetValue(STRESS_VECTOR);
+        const double damage = (*it_elem)->GetValue(DAMAGE_ELEMENT);
+
+        // Compute Normalized Free Energy on that element
+        const double normalized_free_energy = this->CalculateNormalizedFreeEnergy(r_strain_vector,
+                                                                                    r_stress_vector,
+                                                                                    damage,
+                                                                                    r_mat_properties,
+                                                                                    r_geometry);
+
+        for (unsigned int i = 0; i < r_geometry.PointsNumber(); i++) {
+            const int node_id = r_geometry.GetPoint(i).Id();
+            pNodeNormalizedFreeEnergyVector[node_id - 1].NormalizedFreeEnergy += normalized_free_energy;
+            pNodeNormalizedFreeEnergyVector[node_id - 1].NElems += 1;
+        }
+    }
+
+    // Ponderate over the elements coincident on that node
+    for (unsigned int i = 0; i < mNNodes; i++) {
+        pNodeNormalizedFreeEnergyVector[i].NormalizedFreeEnergy /= pNodeNormalizedFreeEnergyVector[i].NElems;
+    }
+
+    // Loop over nodes to assign the variable
+    for (ModelPart::NodeIterator it_node = mrModelPart.NodesBegin(); it_node != mrModelPart.NodesEnd(); ++it_node) {
+        int Id = (*it_node).Id();
+        const double nodal_free_energy = pNodeNormalizedFreeEnergyVector[Id - 1].NormalizedFreeEnergy;
+        double& r_norm = it_node->GetSolutionStepValue(EQUIVALENT_NODAL_STRESS);
+        r_norm = nodal_free_energy;
+
+        if (mCorrectWithDisplacements) {
+            const double r_displ = norm_2(it_node->GetSolutionStepValue(DISPLACEMENT));
```

Comment:
```suggestion
            const double displ = norm_2(r_node.FastGetSolutionStepValue(DISPLACEMENT));
```

Is is not a reference